### PR TITLE
Renames and new UI improves

### PR DESCRIPTION
**Pull request by portellaa**
* Renamed identifier protocol to ReusableView
	* Base class for some cells

### CODE REVIEW

File: Alicerce/Sources/View/CollectionReusableView.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+
+        setupLayout()
+    }
+}
+
+extension CollectionReusableView: View {
+    public func setupLayout() {
+        preconditionFailure("💥 you forgot to override the method? 😱")
```
Comment (p4checo) [inline Alicerce/Sources/View/CollectionReusableView.swift]: I would rephrase these messages to: `💥: Did you forget to override this method? 😱`

Also, I would perhaps change these to `assertionFailure`'s so that it won't crash in Release builds.

This should (hopefully) only crash in Debug builds, anyway, right? 😝

File: Alicerce/Sources/Protocols/View.swift [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-//  ViewCellProtocol.swift
+//  View.swift
 //  Alicerce
 //
 //  Created by Luís Afonso on 16/12/2016.
 //  Copyright © 2016 Mindera. All rights reserved.
 //
 
-public protocol ViewCellProtocol: ViewCellReuseIdentifier {
+public protocol View: ReusableView {
```
Comment (portellaa) [inline Alicerce/Sources/Protocols/View.swift]: Yeap, this protocol is meant to be used by any view, to improve init when we want to do it on code.
The usual use case is `let someWeirdView = UIView() and it works`

**Comment by portellaa** [comment]
Done 💪 🚶

File: Alicerce.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -98,7 +100,9 @@
 			isa = PBXGroup;
 			children = (
 				1BDD79411E03840F00CC1EDA /* Extensions */,
+				1BE256D21E928FFB00942DAC /* View */,
```
Comment (p4checo) [inline Alicerce.xcodeproj/project.pbxproj]: This duplicated reference seems odd. It must have slipped thru during conflict resolution 😝

File: Alicerce.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -142,6 +143,24 @@
 			path = Utils;
 			sourceTree = "<group>";
 		};
+		1BE256D11E928F6300942DAC /* View */ = {
```
Comment (p4checo) [inline Alicerce.xcodeproj/project.pbxproj]: Same as above

File: Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift [hunk 5]
```diff
@@ -10,7 +10,7 @@ import UIKit
 
 public extension UICollectionView {
     
-    public func cell<T: ViewCellProtocol>(`for` indexPath: IndexPath) -> T {
+    public func cell<T: UICollectionView>(`for` indexPath: IndexPath) -> T where T: ReusableView {
```
Comment (p4checo) [inline Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift]: This should be `<T: UICollectionViewCell>` 😉

File: Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift [hunk 6]
```diff
@@ -19,11 +19,12 @@ public extension UICollectionView {
...
     }
     
-    public func register<T: UICollectionViewCell>(_ cellType: T.Type) where T: ViewCellProtocol {
+    public func register<T: UICollectionViewCell>(_ cellType: T.Type) where T: ReusableView {
         register(cellType, forCellWithReuseIdentifier: cellType.reuseIdentifier)
     }
     
-    public func register<T: AnyObject>(_ viewType: T.Type, forSupplementaryViewOfKind kind: String) where T: ViewCellProtocol {
+    public func register<T: UICollectionReusableView>(_ viewType: T.Type, forSupplementaryViewOfKind kind: String)
+        where T: ReusableView {
```
Comment (p4checo) [inline Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift]: I tend to line up `where` clauses at the same indentation level as the `func` keyword. However, it's a matter of preference.

File: Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift [hunk 7]
```diff
@@ -10,7 +10,7 @@ import UIKit
 
 public extension UICollectionView {
     
-    public func cell<T: ViewCellProtocol>(`for` indexPath: IndexPath) -> T {
+    public func cell<T: UICollectionView>(`for` indexPath: IndexPath) -> T where T: ReusableView {
         guard let cell = self.dequeueReusableCell(withReuseIdentifier: T.reuseIdentifier, for: indexPath) as? T else {
             assertionFailure("🔥 Did you forgot to register cell with identifier `\(T.reuseIdentifier)` for type: `\(T.self)`")
```
Comment (p4checo) [inline Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift]: Grammar 👮: Use either "Did you forget to..." or "Forgot to..."

File: Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift [hunk 8]
```diff
@@ -19,11 +19,12 @@ public extension UICollectionView {
         return cell
     }
     
-    public func register<T: UICollectionViewCell>(_ cellType: T.Type) where T: ViewCellProtocol {
+    public func register<T: UICollectionViewCell>(_ cellType: T.Type) where T: ReusableView {
```
Comment (p4checo) [inline Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift]: I don't think we need to specify `public` for all inner vars and methods, since we specify it at a higher scope level. More cases throughout the PR 😉

File: Alicerce/Sources/Protocols/ReusableView.swift [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+public protocol ReusableView: View {
+    static var reuseIdentifier: String { get }
+}
+
+public extension ReusableView where Self: UIView {
+    public static var reuseIdentifier: String { return "\(self)" }
+}
+
+public extension ReusableView where Self: UITableViewCell {
+    public init() {
```
Comment (p4checo) [inline Alicerce/Sources/Protocols/ReusableView.swift]: Perhaps we could have a `style` parameter defaulting to `.default`. What do you say?

File: Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift (hunk 7 above)
Comment (portellaa) [inline Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift]: 🚓

File: Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift (hunk 8 above)
Comment (portellaa) [inline Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift]: Xcode default behaviour, i'm afraid. I don't remind to add it, i'm lazy 😝

Comment (portellaa) [inline Alicerce/Sources/Extensions/UICollectionView+ReusableView.swift]: 🚔

File: Alicerce/Sources/Protocols/ReusableView.swift (hunk 9 above)
Comment (portellaa) [inline Alicerce/Sources/Protocols/ReusableView.swift]: The use case of this method, is to do cell calculations, mostly. It was useful for me in the past, don't know if it will be used yet, but perhaps we can live it for now and remove it later

File: Alicerce.xcodeproj/project.pbxproj (hunk 3 above)
Comment (portellaa) [inline Alicerce.xcodeproj/project.pbxproj]: fixed 😄

File: Alicerce.xcodeproj/project.pbxproj (hunk 4 above)
Comment (portellaa) [inline Alicerce.xcodeproj/project.pbxproj]: same as above 😄

File: Alicerce/Sources/Protocols/ReusableView.swift (hunk 9 above)
Comment (p4checo) [inline Alicerce/Sources/Protocols/ReusableView.swift]: Ok, fair enough 😉 

Also missed that we need that `init()` for `View` conformance 😅
